### PR TITLE
Support gauge metrics in portable mode

### PR DESCRIPTION
**Pull request by ryucc**
Support gauge metrics in portable mode

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoConstants.java [hunk 1]
```diff
@@ -41,6 +41,8 @@ public static final class Urns {
     public static final String FINISH_BUNDLE_MSECS =
         extractUrn(MonitoringInfoSpecs.Enum.FINISH_BUNDLE_MSECS);
     public static final String TOTAL_MSECS = extractUrn(MonitoringInfoSpecs.Enum.TOTAL_MSECS);
+    public static final String USER_LATEST_INT64 = extractUrn(MonitoringInfoSpecs.Enum.USER_LATEST_INT64);
+    public static final String USER_LATEST_DOUBLE = extractUrn(MonitoringInfoSpecs.Enum.USER_LATEST_DOUBLE);
```
Comment (alnzng) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoConstants.java]: Do we need this double type somewhere? Just according to this pr, sees we only use the int64.

**Review by alnzng — COMMENTED** [review]
overall LGTM, just one minor comment

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoConstants.java (hunk 1 above)
Comment (ryucc) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoConstants.java]: updated

**Comment by mynameborat** [comment]
Can we add unit tests?

**Comment by ryucc** [comment]
@mynameborat I'll check

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java [hunk 2]
```diff
@@ -357,6 +367,14 @@ public Map<String, ByteString> getMonitoringData(ShortIdMap shortIds) {
         builder.put(shortId, encodeInt64Distribution(metricUpdate.getUpdate()));
       }
     }
+    for (MetricUpdate<org.apache.beam.runners.core.metrics.GaugeData> metricUpdate :
```
Comment (Sanil15) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: nit: s/org.apache.beam.runners.core.metrics.GaugeData/Guage

**Review by Sanil15 — CHANGES_REQUESTED** [review]
nitpick, rest LGTM

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java (hunk 2 above)
Comment (ryucc) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Good catch. Somehow it got autoformatted..

Comment (ryucc) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Wait... I don't understand this comment.

Comment (ryucc) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: oh sed comment.

Comment (ryucc) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: fixed

**Comment by ryucc** [comment]
@mynameborat good catch. Fixed!

**Review by alnzng — APPROVED** [review]

**Review by Sanil15 — APPROVED** [review]

**Pull request merged**